### PR TITLE
Add user user attributes funcs in node; high five logic

### PR DESCRIPTION
**Pull request by boyan-d**

### CODE REVIEW

File: universe/user/message_loop.go [hunk 1]
```diff
@@ -171,3 +175,80 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	_, err := universe.GetNode().UpsertUserUserAttribute(entry.NewUserUserAttributeID(
+		entry.NewAttributeID(universe.GetSystemPluginID(), universe.Attributes.User.HighFive.Name), u.id, targetID), modifyFn)
+	if err != nil {
+		return errors.New("Could not upsert high-five user user attribute")
+	}
+
+	target, found := u.space.GetUser(targetID, false)
+	if !found {
+		posbus.NewSimpleNotificationMsg(
+			posbus.DestinationReact, posbus.NotificationTextMessage, 0, "Receiving user not found")
```
Comment (RamsesDijkstra) [inline universe/user/message_loop.go]: Maybe change this to "Target user not found" instead

File: universe/user/message_loop.go [hunk 2]
```diff
@@ -171,3 +175,80 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	}
+
+	effect.SetEffect(0, effectsEmitterID, *u.pos, target.GetPosition(), 1001)
+	u.world.Send(effect.WebsocketMessage(), false)
+	go u.SendHighFiveStats(&target)
+
+	return nil
+}
+
+func (u *User) SendHighFiveStats(target *universe.User) error {
```
Comment (RamsesDijkstra) [inline universe/user/message_loop.go]: i think this function needs to go to universe/user/stats.go

File: universe/user/message_loop.go [hunk 3]
```diff
@@ -171,3 +175,80 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+		return current, nil
+	}
+
+	_, err := universe.GetNode().UpsertUserUserAttribute(entry.NewUserUserAttributeID(
+		entry.NewAttributeID(universe.GetSystemPluginID(), universe.Attributes.User.HighFive.Name), u.id, targetID), modifyFn)
+	if err != nil {
+		return errors.New("Could not upsert high-five user user attribute")
+	}
+
+	target, found := u.space.GetUser(targetID, false)
```
Comment (cnaize) [inline universe/user/message_loop.go]: do you need exectly space? maybe world?

File: universe/user/message_loop.go [hunk 4]
```diff
@@ -171,3 +175,80 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	uProfile := u.GetProfile()
+
+	msg := struct {
+		SenderID   string `json:"senderId"`
+		ReceiverID string `json:"receiverId"`
+		Message    string `json:"message"`
+	}{
+		SenderID:   u.id.String(),
+		ReceiverID: targetID.String(),
+		Message:    *uProfile.Name,
```
Comment (cnaize) [inline universe/user/message_loop.go:226]: name maybe be nil

File: universe/user/message_loop.go [hunk 5]
```diff
@@ -171,3 +175,80 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	if !ok {
+		return errors.Errorf("Could not get effects emitter attribute")
+	}
+
+	effectsEmitterID := utils.GetFromAnyMap(*effectsEmitterMap, universe.Attributes.World.EffectsEmitter.Key, uuid.Nil)
+	if effectsEmitterID == uuid.Nil {
+		return errors.Errorf("Failed to get effects emitter ID from map")
+	}
+
+	effect.SetEffect(0, effectsEmitterID, *u.pos, target.GetPosition(), 1001)
```
Comment (cnaize) [inline universe/user/message_loop.go]: you don't lock the method, that's why it's dangerous to use internal variables directly

File: universe/user/message_loop.go [hunk 6]
```diff
@@ -141,13 +145,13 @@ func (u *User) InteractionHandler(m *posbus.TriggerInteraction) error {
 			)
 		}
 		return nil
+	case posbus.TriggerHighFive:
+		if err := u.HandleHighFive(m); err != nil {
+			u.log.Warn(errors.WithMessage(err, "InteractionHandler: trigger high fives: failed to handle high five"))
+		}
```
Comment (cnaize) [inline universe/user/message_loop.go:151]: forgot to return nil

File: universe/user/message_loop.go [hunk 7]
```diff
@@ -171,3 +176,82 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
 
 	return u.GetWorld().Send(msg.WebsocketMessage(), true)
 }
+
+func (u *User) HandleHighFive(m *posbus.TriggerInteraction) error {
```
Comment (cnaize) [inline universe/user/message_loop.go:180]: you are using private variables directly without any locks

File: universe/user/message_loop.go [hunk 8]
```diff
@@ -171,3 +176,78 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	effectsEmitterMap, ok := universe.GetNode().GetSpaceAttributeValue(entry.NewAttributeID(universe.GetSystemPluginID(), universe.Attributes.World.EffectsEmitter.Name))
+	if !ok {
+		return errors.Errorf("Could not get effects emitter attribute")
+	}
+
+	effectsEmitterID := utils.GetFromAnyMap(*effectsEmitterMap, universe.Attributes.World.EffectsEmitter.Key, uuid.Nil)
+	if effectsEmitterID == uuid.Nil {
+		return errors.Errorf("Failed to get effects emitter ID from map")
+	}
+
```
Comment (cnaize) [inline universe/user/message_loop.go:247]: here again

File: universe/user/message_loop.go [hunk 9]
```diff
@@ -171,3 +176,78 @@ func (u *User) LockObject(msg *posbus.SetObjectLockState) error {
...
+	effect.SetEffect(0, effectsEmitterID, u.GetPosition(), target.GetPosition(), 1001)
+	u.GetWorld().Send(effect.WebsocketMessage(), false)
+	go u.SendHighFiveStats(&target)
+
```
Comment (cnaize) [inline universe/user/message_loop.go:251]: I think you need ActualPosition

**Pull request merged**